src/classes/GS_NonprofitTrialOrgService.cls

```diff
@@ -0,0 +1,173 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @date 2020
+ * @group Getting Started
+ * @author Salesforce.org
+ * @description Service class for getting the Licensing Status of the org
+ */
+public with sharing class GS_NonprofitTrialOrgService {
+
+    private static final String STATIC_RESOURCE_NAME = 'gsApplicationConfiguration';
+    private static final String APPLICATION_DATE = 'P10_Application_Date';
+    private static final String SANDBOX_PARAMETER = 'Sandbox';
+    private static final String APPLICATION_PARAMETER = 'Application';
+    private static final String TRIAL_EXPIRATION_PARAMETER = 'TrialExpiration';
```

Comment:
Rather than deserialize this structure into a map, I recommend using a simple private inner class:
```
private class TestingConfig {
    public Date applicationDate;
    public Boolean isSandbox;
    public String application;
    public Date trialExpiration;
}
```
Then just deserialize:
```
TestingConfig testConfig = (TestingConfig)JSON.deserialize(staticResources[0].Body.toString(), TestingConfig.class);
```

And you can reference properties without type casting:
`Date applicationDate = testConfig.applicationDate;`

This would be significantly easier to follow in the code as well as debug and test.